modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/AbstractPageMemoryTableStorage.java

```diff
@@ -185,7 +212,7 @@ protected void close(boolean destroy) throws StorageException {
             AbstractPageMemoryMvPartitionStorage partition = mvPartitions.getAndUpdate(i, p -> null);
 
             if (partition != null) {
-                closeables.add(destroy ? partition::destroy : partition::close);
+                closeables.add(destroy ? () -> destroyMvPartitionStorage(partition) : partition::close);
```

Comment:
Is it ok that we don't wait for any partition to be actually destroyed (as we just ignore the returned futures)?

Reply:
You're right, I need to fix this.